### PR TITLE
Overhaul find command

### PR DESCRIPTION
**Pull request by eman-kom**
Add g/ flag to simulate AND operators
Chain g/ flags to simulate OR operators
Modified some test cases
Enabled assertions in build.gradle
`jobid:` is a special keyword used to find for job id
`progress:` is a special keyword used to find for progress
Add contains method in person class to check if person contains keyword
Update UG

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -35,48 +27,22 @@ public FindCommand parse(String args) throws ParseException {
...
         }
 
-        ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_APPLICATION);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_GROUP);
 
-        HashMap<Prefix, String> searchDict = new HashMap<>();
+        if (argMultimap.getValue(PREFIX_GROUP).isPresent()) {
+            List<String> terms = argMultimap.getAllValues(PREFIX_GROUP);
+            boolean isEmpty = terms.stream().anyMatch(x -> x.equals(""));
```
Comment (brigittesantoso) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:34]: Good use of any match function here!

File: src/main/java/seedu/address/model/person/Person.java [hunk 2]
```diff
@@ -95,6 +95,47 @@ public boolean equals(Object other) {
                 && otherPerson.getApplications().equals(getApplications());
     }
 
+    /**
+     * Checks if this person contains the given keyword.
+     *
+     * @param term The search term.
+     * @return A boolean checking whether this person has the keyword or not.
+     */
+    public boolean contains(String term) {
```
Comment (brigittesantoso) [inline src/main/java/seedu/address/model/person/Person.java:104]: If you want, could possibly split this method into 2 as it is rather long!

**Review by brigittesantoso — APPROVED** [review]
Overall LGTM

File: src/main/java/seedu/address/model/person/Person.java (hunk 2 above)
Comment (eman-kom) [reply src/main/java/seedu/address/model/person/Person.java:104]: fixed

**Pull request merged**